File: 392/is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        int ti=0;
        for(int si=0; si<s.length(); si++){
            char sc=s.charAt(si);
            while(ti<t.length()){
                char tc=t.charAt(ti);
                if(sc==tc) break;
                ti++;
            }
            if(si==s.length()-1 && ti==t.length()-1) return true;
            if(ti==t.length()) return false;

            ti++;
        }
        return true;
    }
}

File: 392/is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        i = 0 # for t[i]
        for c in s:
            while i<len(t) and t[i] != c:
                i += 1 # 無法正確比對到, 就換下一個
            if i == len(t): # 走到超過範圍, 就表示失敗
                return False
            i += 1 # 迴圈最後, 記得 i增加, 以準備下一輪的比較
        return True
        
